project/src/main/java/com/google/sps/data/DataAccessor.java

```diff
@@ -49,41 +58,89 @@ public DataAccessor() {
     Query query = new Query(USER_ENTITY_KIND).setFilter(userIdFilter).setKeysOnly();
-    return datastoreService.prepare(query)
+    return !datastoreService.prepare(query)
         .asList(FetchOptions.Builder.withDefaults())
-        .size() > 0;
+        .isEmpty();
   }
 
   /**
   * Registers the user in our system, by adding an entity that represents them to datastore.
   *
-  * @param userId the id of the user that we want to register to our system
+  * @param userId the ID of the user that we want to register to our system
   */
   public void registerUser(String userId) {
-    checkArgument(!isNullOrEmpty(userId), "User ID may not be null or empty");
+    checkArgument(!isNullOrEmpty(userId), INVALID_USER_MSG);
     checkArgument(!isRegistered(userId), "User already registered.");
     Entity userEntity = new Entity(USER_ENTITY_KIND, userId);
     datastoreService.put(userEntity);
   }
 
+  /**
+  * Updates the database with information from the user feedback. Each entity represents the
+  * relations between the user and a single place that was recommended to them. So for each user
+  * feedback, multiple entities are generated, to represent each place seperately.
+  *
+  * @param feedback a UserFeedback that holds all the information about the user feedback.
+  */
+  public void updateUserFeedback(UserFeedback feedback) {
+    for (String place : feedback.recommendedPlaces()) {
+      Entity recommendationEntity = new Entity(RECOMMENDATION_ENTITY_KIND);
+      recommendationEntity.setProperty(USER_ID_PROPERTY, feedback.userId());
+      recommendationEntity.setProperty(PLACE_ID_PROPERTY, place);
+      recommendationEntity.setProperty(CHOSEN_PROPERTY,
+          feedback.chosenPlace().isPresent() && feedback.chosenPlace().get().equals(place));
+      recommendationEntity.setProperty(TRY_AGAIN_PROPERTY, feedback.userTriedAgain());
+      recommendationEntity.setProperty(TIME_PROPERTY, feedback.feedbackTimeInMillis());
+      datastoreService.put(recommendationEntity);
+    }
+  }
+
+  /**
+  * Approaches the database and gets the places that were recommended to the user in the past.
+  * There is a possibility to get only places that the user chose, according to their feedback.
+  *
+  * @param userId the user we want the information about.
+  * @param getOnlyPlacesUserChose if this is true, the returned places would consist only of places
+  *                               that the user chose, according to their feedback.
+  * @return the IDs of the places that the user received recommendations about in the past.
+  */
+  public ImmutableList<String> getPlacesRecommendedToUser(String userId,
+      boolean getOnlyPlacesUserChose) {
+    checkArgument(!isNullOrEmpty(userId), INVALID_USER_MSG);
+    Filter userIdFilter =
+        new Query.FilterPredicate(USER_ID_PROPERTY, FilterOperator.EQUAL, userId);
+    Filter chosenPlacesFilter =
+        new Query.FilterPredicate(CHOSEN_PROPERTY, FilterOperator.EQUAL, true);
```

Comment:
Make `chosenPlacesFilter` a static final variable of the class?

Reply:
sure, done